alacritty_terminal/src/term/mod.rs

```diff
@@ -650,24 +651,35 @@ impl SizeInfo {
         (self.padding_y + (self.height - 2. * self.padding_y) % self.cell_height) as usize
     }
 
+    /// Update the number of columns/lines based on window dimensions.
+    #[inline]
+    pub fn update_dimensions(&mut self, reserved_lines: usize) {
+        let lines = (self.height - 2. * self.padding_y) / self.cell_height;
+        let free_lines = (lines as usize).saturating_sub(reserved_lines);
+        self.screen_lines = Line(max(free_lines, MIN_SCREEN_LINES));
+
+        let cols = (self.width - 2. * self.padding_x) / self.cell_width;
+        self.cols = Column(max(cols as usize, MIN_COLS))
+    }
+
     /// Check if coordinates are inside the terminal grid.
     ///
-    /// The padding is not counted as part of the grid.
+    /// The padding, message bar or search are not counted as part of the grid.
```

Comment:
They are not counted, since those are not part of the alacritty_terminal in particular, and such comment is confusing, since message bar is ui thing from alacritty. I'd suggest to remove this addition.

Reply:
Ah, you mean `alacritty_terminal` doesn't know about them. That's a fair point. Though previously they were included so that's why I made the change.

It's a bit difficult because it's a very UI oriented structure.

Reply:
Padding is a part of SizeInfo and the contains point checks for padding in particular, however we know nothing about search and message bar, so remove those two and keep padding.

Reply:
Of course we know about these, they're explicitly removed with a dedicated `reserve_lines` method. We have explicit support **just for these things**. Trying to hide something for the purpose of hiding it is pointless. Documenting the reason why it exists, even if it relates to external factors, just makes things easier for everyone.

There's no reason why we would lie about this.